pkg/network/protocols/events/configuration.go

```diff
@@ -8,45 +8,194 @@
+	mapName := eventMapName(proto)
+	ringBufferSize := toPowerOf2(numCPUs * defaultPerfEventBufferSize)
+	rb := &manager.RingBuffer{
+		Map: manager.Map{Name: mapName},
+		RingBufferOptions: manager.RingBufferOptions{
+			RingBufferSize: ringBufferSize,
+			RecordHandler:  handler.RecordHandler,
+			RecordGetter:   handler.RecordGetter,
+		},
+	}
+
+	o.MapSpecEditors[mapName] = manager.MapSpecEditor{
+		Type:       ebpf.RingBuf,
+		MaxEntries: uint32(ringBufferSize),
+		KeySize:    0,
+		ValueSize:  0,
+		EditorFlag: manager.EditType | manager.EditMaxEntries | manager.EditKeyValue,
+	}
+
+	m.RingBuffers = append(m.RingBuffers, rb)
+	setHandler(proto, handler)
+}
+
+func configureBatchMaps(proto string, o *manager.Options, numCPUs int) {
 	if o.MapSpecEditors == nil {
 		o.MapSpecEditors = make(map[string]manager.MapSpecEditor)
 	}
 
 	o.MapSpecEditors[proto+batchMapSuffix] = manager.MapSpecEditor{
-		MaxEntries: uint32(onlineCPUs * batchPagesPerCPU),
+		MaxEntries: uint32(numCPUs * batchPagesPerCPU),
 		EditorFlag: manager.EditMaxEntries,
 	}
 }
 
-func getHandler(proto string) *ddebpf.PerfHandler {
+func eventMapName(proto string) string {
+	return proto + eventsMapSuffix
+}
+
+// noopInstruction is used for the purposes of eBPF patching (see comments
+// below)
+var noopInstruction = asm.Mov.Reg(asm.R1, asm.R1)
+
+// removeRingBufferHelperCalls is called only in the context of kernels that
+// don't support ring buffers. our eBPF code looks more or less like the
+// following:
+//
+//	if (ring_buffers_supported) {
+//	    bpf_ringbuf_output();
+//	} else {
+//	    bpf_perf_event_output();
+//	}
+//
+// where `ring_buffers_supported` is an injected constant. The code above seems
+// to work on the vast majority of kernel versions due to dead code elimination
+// by the verifier, so for kernels that don't support ring buffers
+// (ring_buffers_supported=0) we only see the perf event helper call when doing
+// a program dump:
+//
+// bpf_perf_event_output();
+//
+// *However* in some instances this is not working on 4.14, so here we
+// essentially replace `bpf_ringbuf_output` helper calls by a noop operation so
+// they don't result in verifier errors even when deadcode elimination fails.
+func removeRingBufferHelperCalls(m *manager.Manager) {
+	m.InstructionPatchers = append(m.InstructionPatchers, func(m *manager.Manager) error {
+		progs, err := m.GetProgramSpecs()
+		if err != nil {
+			return err
+		}
+
+		for _, p := range progs {
+			iter := p.Instructions.Iterate()
+			for iter.Next() {
+				ins := iter.Ins
+				if ins.IsBuiltinCall() && ins.Constant == int64(asm.FnRingbufOutput) {
+					*ins = noopInstruction
+				}
+			}
+		}
+
+		return nil
+	})
+}
+
+func alreadySetUp(proto string, m *manager.Manager) bool {
+	// check if we already have configured this perf map this can happen in the
+	// context of a failed program load succeeded by another attempt
+	mapName := eventMapName(proto)
+	for _, perfMap := range m.PerfMaps {
```

Comment:
nit - you can use `slices.ContainsFunc`

Reply:
IMO I don't think this would improve the code in any way, so I'm favoring leaving it unchanged